crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -524,28 +505,36 @@ impl Faucet for SimpleFaucet {
     ) -> Result<FaucetReceipt, FaucetError> {
         info!(?recipient, uuid = ?id, "Getting faucet requests");
 
-        let (digest, coin_ids, sent_amounts) = self.transfer_gases(amounts, recipient, id).await?;
-        if coin_ids.len() != amounts.len() {
-            error!(
-                uuid = ?id, ?recipient,
-                "Requested {} coins but got {}",
-                amounts.len(),
-                coin_ids.len()
-            );
-        }
+        let (digest, coin_ids) = self.transfer_gases(amounts, recipient, id).await?;
 
         info!(uuid = ?id, ?recipient, ?digest, "PaySui txn succeeded");
-        Ok(FaucetReceipt {
-            sent: coin_ids
-                .iter()
-                .zip(sent_amounts)
```

Comment:
Heads up, this logic was already broken since you are not guaranteed to get the coin ids back in order.